File: extension/icons/generate-icons.py
from PIL import Image, ImageDraw
import os

def create_icon(size):
    """Create a modern shield icon with gradient background"""
    # Create image with gradient background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Draw gradient background (purple to pink)
    for y in range(size):
        # Calculate color gradient
        ratio = y / size
        r = int(99 + (236 - 99) * ratio)
        g = int(102 + (72 - 102) * ratio)
        b = int(241 + (153 - 241) * ratio)
        draw.line([(0, y), (size, y)], fill=(r, g, b, 255))
    
    # Draw shield shape
    center_x = size // 2
    center_y = size // 2
    shield_size = int(size * 0.6)
    
    # Shield points
    top = center_y - int(shield_size * 0.4)
    bottom = center_y + int(shield_size * 0.5)
    left = center_x - int(shield_size * 0.35)
    right = center_x + int(shield_size * 0.35)
    
    # Create shield polygon
    shield_points = [
        (center_x, top),
        (right, top + int(shield_size * 0.15)),
        (right, center_y + int(shield_size * 0.1)),
        (center_x, bottom),
        (left, center_y + int(shield_size * 0.1)),
        (left, top + int(shield_size * 0.15))
    ]
    
    # Draw white shield
    draw.polygon(shield_points, fill=(255, 255, 255, 240))
    
    # Add checkmark for larger sizes
    if size >= 48:
        line_width = max(2, size // 16)
        check_color = (99, 102, 241, 255)  # Purple
        
        # Checkmark points
        p1 = (center_x - int(shield_size * 0.15), center_y)
        p2 = (center_x - int(shield_size * 0.05), center_y + int(shield_size * 0.15))
        p3 = (center_x + int(shield_size * 0.2), center_y - int(shield_size * 0.15))
        
        # Draw checkmark
        draw.line([p1, p2], fill=check_color, width=line_width)
        draw.line([p2, p3], fill=check_color, width=line_width)
    
    return img

def main():
    """Generate all icon sizes"""
    sizes = [16, 48, 128]
    
    for size in sizes:
        print(f"Generating icon{size}.png...")
        icon = create_icon(size)
        icon.save(f'icon{size}.png', 'PNG')
        print(f"✅ Generated icon{size}.png")
    
    print("🎉 All icons generated successfully!")

if __name__ == "__main__":
    main()

File: extension/icons/generate-icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

function drawIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Background - Soft gradient (simulate with solid color for simplicity)
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#6366f1');
    gradient.addColorStop(0.5, '#8b5cf6');
    gradient.addColorStop(1, '#ec4899');
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 0, size, size);
    
    // Shield shape - Modern minimal
    const centerX = size / 2;
    const centerY = size / 2;
    const shieldSize = size * 0.6;
    
    ctx.fillStyle = 'rgba(255, 255, 255, 0.95)';
    ctx.beginPath();
    
    // Draw shield
    const top = centerY - shieldSize * 0.4;
    const bottom = centerY + shieldSize * 0.5;
    const left = centerX - shieldSize * 0.35;
    const right = centerX + shieldSize * 0.35;
    
    ctx.moveTo(centerX, top);
    ctx.lineTo(right, top + shieldSize * 0.15);
    ctx.lineTo(right, centerY + shieldSize * 0.1);
    ctx.quadraticCurveTo(right, bottom - shieldSize * 0.2, centerX, bottom);
    ctx.quadraticCurveTo(left, bottom - shieldSize * 0.2, left, centerY + shieldSize * 0.1);
    ctx.lineTo(left, top + shieldSize * 0.15);
    ctx.closePath();
    ctx.fill();
    
    // Add checkmark (if size is large enough)
    if (size >= 48) {
        ctx.strokeStyle = '#6366f1';
        ctx.lineWidth = size / 16;
        ctx.lineCap = 'round';
        ctx.lineJoin = 'round';
        
        ctx.beginPath();
        ctx.moveTo(centerX - shieldSize * 0.15, centerY);
        ctx.lineTo(centerX - shieldSize * 0.05, centerY + shieldSize * 0.15);
        ctx.lineTo(centerX + shieldSize * 0.2, centerY - shieldSize * 0.15);
        ctx.stroke();
    }
    
    return canvas;
}

// Generate icons
const sizes = [16, 48, 128];
sizes.forEach(size => {
    const canvas = drawIcon(size);
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(`icon${size}.png`, buffer);
    console.log(`✅ Generated icon${size}.png`);
});

console.log('🎉 All icons generated successfully!');
